src/index.ts

```diff
@@ -0,0 +1,177 @@
+}
+
+
+// 15 days, in milliseconds
+const PR_EXPIRY_MS = 15 * 24 * 60 * 60 * 1000;
+
+// Github API client
+const gh_client = new github.GithubClient(functions.config().github.token);
+
+// Mailgun Email client
+const email_client = new email.EmailClient(
+  functions.config().mailgun.key,
+  functions.config().mailgun.domain
+);
+
+// Handler for Github issues
+const issue_handler = new issues.IssueHandler(
+  gh_client,
+  email_client,
+  bot_config
+);
+
+// Handler for Github pull requests
+const pr_handler = new pullrequests.PullRequestHandler(
+  gh_client,
+  email_client,
+  bot_config
+);
+
+// Handler for Cron jobs
+const cron_handler = new cron.CronHandler(gh_client);
+
+/**
+ * Function that responds to Github events (HTTP webhook).
+ */
+exports.githubWebhook = functions.https.onRequest((request, response) => {
+  // Get event and action;
+  const event = request.get("X-Github-Event");
+  const action = request.body.action;
+
+  // Get repo and sender
+  const repo = request.body.repository;
+  const sender = request.body.sender;
+
+  // Confirm that there is some event
+  if (!event) {
+    console.log("No github event");
+    response.send("Fail: no event.");
+    return;
+  } else {
+    // Log some basic info
+    console.log("===========================START============================");
+    console.log(`Event: ${event}/${action}`);
+    if (repo) {
+      console.log("Repository: " + repo.full_name);
+    }
+    if (sender) {
+      console.log("Sender: " + sender.login);
+    }
+    console.log("===========================END=============================");
+  }
+
+  // Handle the event appropriately
+  let handlePromise: Promise<any>;
+  const issue = request.body.issue;
+
+  switch (event) {
+    case GithubEvent.ISSUE:
+      handlePromise = issue_handler.handleIssueEvent(
+        request.body,
+        action,
+        issue,
+        repo,
+        sender
+      );
+      break;
+    case GithubEvent.ISSUE_COMMENT:
+      const comment = request.body.comment;
+      handlePromise = issue_handler.handleIssueCommentEvent(
+        request.body,
+        action,
+        issue,
+        comment,
+        repo,
+        sender
+      );
+      break;
+    case GithubEvent.PULL_REQUEST:
+      const pr = request.body.pull_request;
+      handlePromise = pr_handler.handlePullRequestEvent(
+        request.body,
+        action,
+        pr,
+        repo,
+        sender
+      );
+      break;
+    default:
+      response.send(`Unknown event: ${event}`);
+      return;
+  }
+
+  // Wait for the promise to resolve the HTTP request
+  handlePromise
+    .then(res => {
+      response.send("OK!");
+    })
+    .catch(e => {
+      response.send("Error!");
+    });
+});
+
+/**
+ * Function that responds to pubsub events sent via an AppEngine crojob.
+ */
+exports.timedCleanup = functions.pubsub.topic("cleanup").onPublish(event => {
+  // TODO(abehaskins): I think this code is broken, where"s this.bot_config?
```

Comment:
This function references `this.bot_config` which I think might be coming from global? Not sure how this could work